### PR TITLE
doc/glossary: add AWS links to bucket info

### PR DESCRIPTION
**Pull request by zdover23**
Add links to AWS's documentation of buckets, in accordance with Casey Bodley's suggestion here:
https://github.com/ceph/ceph/pull/50221#discussion_r1115900879

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/glossary.rst [hunk 1]
```diff
@@ -16,13 +16,21 @@
 
         Bucket
                 In the context of :term:`RGW`, a bucket is a group of objects.
-                Buckets are grouped into zones, and zones are grouped into
-                zonegroups. In an filesystem-based analogy in which objects are
-                the counterpart of files, buckets are the counterpart of
+                In an filesystem-based analogy in which objects are the
```
Comment (anthonyeleven) [inline doc/glossary.rst:19]: This seems stilted, but for sure s/an/a/

File: doc/glossary.rst [hunk 2]
```diff
@@ -16,13 +16,21 @@
...
+                counterpart of files, buckets are the counterpart of
                 directories. :ref:`Multisite sync
                 policies<radosgw-multisite-sync-policy>` can be set on buckets,
                 to provide fine-grained control of data movement from one zone
-                to another zone.
+                to another zone. 
+                
+                The concept of the bucket has been taken from
+                AWS. See also `the AWS page on creating
```
Comment (anthonyeleven) [inline doc/glossary.rst:27]: AWS S3

**Review by anthonyeleven — APPROVED** [review]
See comments, nonblocking to avoid handshaking

File: doc/glossary.rst (hunk 1 above)
Comment (zdover23) [reply doc/glossary.rst:19]: Yep. That's an artifact of a sentence that originally lacked the adjective "filesystem-based", a sentence in which the indefinite article was yoked to its noun.

File: doc/glossary.rst (hunk 2 above)
Comment (zdover23) [reply doc/glossary.rst]: Noted and accepted.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/50247 - Quincy backport
https://github.com/ceph/ceph/pull/50248 - Pacific backport

**Pull request merged**